### PR TITLE
there's only one db now, get it from ENV

### PR DESCRIPTION
**Pull request by longrunningprocess**
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/914)

### CODE REVIEW

**Review by longrunningprocess — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/914)* status: 0 of 16 files reviewed, 7 unresolved discussions (waiting on @longrunningprocess)

*[a discussion](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmt0f2-myOAP2RcitP:-MXmt0f2-myOAP2RcitQ:b-stkod4) (no related file):*
I was able to run the app locally, run db-reset and run the unit tests successfully locally so I assume these changes are good but please let me know if there are other things I could be doing to exercise more of the app.

The E2E test were flaky locally so we'll see for sure on TC about those.

---
___
*[docker/app-for-e2e/Dockerfile, line 4 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmr_deBqAiuyFggNAa:-MXmr_deBqAiuyFggNAb:beu6y4j) ([raw file](https://github.com/sillsdev/web-languageforge/blob/71fbe5dab97744574ff7829c2881037e50c43a5a/docker/app-for-e2e/Dockerfile#L4)):*
> ```Dockerfile
> 
> COPY src/config.php /var/www/src/config.php
> RUN echo "define('TestMode', true);" >> /var/www/src/config.php
> ```

I saw one use of this 'TestMode' flag in the code but I"m not sure what else it's driving...this would be better if wherever this is used, it was based off `ENVIRONMENT == 'testing'` or something like that...I don't like the idea there's conditional logic sprinkled around for determining where the code is being executed.  We can talk about it more but usually we can ensure code is only ever run in the environment it was built for.
___
*[src/config.php.fortest, line 0 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsAY42wFmbsqCP9jy:-MXmsAY42wFmbsqCP9jz:b-bpjbbc) ([raw file](https://github.com/sillsdev/web-languageforge/blob/71fbe5dab97744574ff7829c2881037e50c43a5a/src/config.php.fortest#L0)):*
the only effective diff between this one and the basic config was the presence of the `TestMode` flag.
___
*[src/config.php.live, line 0 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsLplB8rRW36N2hND:-MXmsLpmBd4JaSCqq_eQ:b-4k2jct) ([raw file](https://github.com/sillsdev/web-languageforge/blob/71fbe5dab97744574ff7829c2881037e50c43a5a/src/config.php.live#L0)):*
I'm assuming this is not needed anymore (same as the "fortest" version.  All of these environmentally changing values will be fed in via ENV vars.
___
*[src/Api/Library/Shared/CLI/RunClass.php, line 35 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsaE30yLLE3lUaN5y:-MXmsaE30yLLE3lUaN5z:b-sdzviy) ([raw file](https://github.com/sillsdev/web-languageforge/blob/6b5a12e0cd36148076784927d5aaf2a1eeffd820/src/Api/Library/Shared/CLI/RunClass.php#L35)):*
> ```PHP
> if ($isTest) {
>     define('TestMode', true);
>     define('SF_DATABASE', 'scriptureforge_test');
> ```

not needed anymore.
___
*[test/app/setupTestEnvironment.php, line 36 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsrgyCpzzGTYXL5OP:-MXmsrgyCpzzGTYXL5OQ:b-1crk6g) ([raw file](https://github.com/sillsdev/web-languageforge/blob/6b5a12e0cd36148076784927d5aaf2a1eeffd820/test/app/setupTestEnvironment.php#L36)):*
> ```PHP
> $site = $website->base;
> 
> if (strpos(SF_DATABASE, '_test') === false ) {
> ```

not sure this is even relevant...
___
*[test/common/MongoTestEnvironment.php, line 40 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsx3H53L2v1HU0Xo8:-MXmsx3H53L2v1HU0Xo9:brecmdr) ([raw file](https://github.com/sillsdev/web-languageforge/blob/6b5a12e0cd36148076784927d5aaf2a1eeffd820/test/common/MongoTestEnvironment.php#L40)):*
> ```PHP
>     /**
>      * Removes all the collections from the mongo database.
>      * Hopefully this is only ever called on the scriptureforge_test database.
> ```

I don't think this is possible.

**Comment by longrunningprocess** [comment]
don't forget the following after PR is approved:

- [ ]  bump build-n-publish-images to appropriate commit
- [ ]  add ENV vars to k8s and redeploy

**Review by megahirt — CHANGES_REQUESTED** [review]
Awesome job!  I love that you renamed the database constant to remove the `SF_`

To me the only real thing you missed was the fact that we have a `scriptureforge_test` database that sits alongside the real `scriptureforge` database.  These two databases should only exist beside each other on a developer machine or CI agent, otherwise the production instances should only have a `scriptureforge` database.

> Reviewed 12 of 16 files at r1, 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-:-MY3vT-28QozhLz5so8t:b-uezjnd)* status: all files reviewed, 9 unresolved discussions (waiting on @longrunningprocess)

*[a discussion](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmt0f2-myOAP2RcitP:-MY3w2vLA7ARojac8pjz:b-yza9bk) (no related file):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

I was able to run the app locally, run db-reset and run the unit tests successfully locally so I assume these changes are good but please let me know if there are other things I could be doing to exercise more of the app.

The E2E test were flaky locally so we'll see for sure on TC about those.
</blockquote></details>

Yes, the tests should run with this code, however without distinguishing between the scriptureforge and scriptureforge_test databases you won't be able to run up user data and run tests (the tests would clobber the user data)

---
___
*[docker/docker-compose.yml, line 136 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MY3o0R-1EkLYDzXVtij:-MY3o0R-1EkLYDzXVtik:b-krhloz) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L136)):*
> ```YAML
>       - WAIT_HOSTS=db:27017, mail:25, selenium:4444
>       - ENVIRONMENT=development
>       - DATABASE=scriptureforge
> ```


I'm actually not certain that the DATABASE line is necessary.  Is it used anywhere in this container?  Or did you just add it for completeness?
___
*[docker/docker-compose.yml, line 164 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MY3nws6EAM2-30nQCC6:-MY3nws6EAM2-30nQCC7:b4kk52g) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L164)):*
> ```YAML
>       - "LDAPI_BASE_URL=http://ld-api:3000/api/v2/"
>       - ENVIRONMENT=development
>       - DATABASE=scriptureforge
> ```

There is a separate database for running tests, `scriptureforge_test`.  I suggest an edit in github review.
___
*[docker/docker-compose.yml, line 181 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MY3njSs2qNKzP0SbRU7:-MY3njSs2qNKzP0SbRU8:b4kk52g) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L181)):*
> ```YAML
>       - WAIT_HOSTS=db:27017, mail:25
>       - ENVIRONMENT=development
>       - DATABASE=scriptureforge
> ```

There is a separate database for running tests, `scriptureforge_test`.  I suggest an edit in github review.
___
*[docker/app-for-e2e/Dockerfile, line 4 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmr_deBqAiuyFggNAa:-MY3msy_BeFH2Zml6W7W:b-adw3ro) ([raw file](https://github.com/sillsdev/web-languageforge/blob/71fbe5dab97744574ff7829c2881037e50c43a5a/docker/app-for-e2e/Dockerfile#L4)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

I saw one use of this 'TestMode' flag in the code but I"m not sure what else it's driving...this would be better if wherever this is used, it was based off `ENVIRONMENT == 'testing'` or something like that...I don't like the idea there's conditional logic sprinkled around for determining where the code is being executed.  We can talk about it more but usually we can ensure code is only ever run in the environment it was built for.
</blockquote></details>

Since you've brought it up, I looked and only found 3 references to `TestMode`, all in `CommunicateHelper.php`.  The three instances don't seem to add value to me, and I wonder if we can simply remove the variable from the code and keep just "not test mode" case.  I suspect that we will be able to run the unit and e2e tests just fine without it.

So in conclusion, just delete references to TestMode all together.  It looks like the test config also defined the database name, which I've addressed above.  Since the database name is coming in via environment variable, then maybe you can really get rid of the config.fortest file all together - that would be sweet!
___
*[src/config.php.fortest, line  at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsAY42wFmbsqCP9jy:-MY3pJg-0iScnRBZxp0B:b2cj2s3) ([raw file](https://github.com/sillsdev/web-languageforge/blob/71fbe5dab97744574ff7829c2881037e50c43a5a/src/config.php.fortest#L)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

the only effective diff between this one and the basic config was the presence of the `TestMode` flag.
</blockquote></details>

looks good - I agree with getting rid of it.
___
*[src/config.php.live, line  at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsLplB8rRW36N2hND:-MY3pQdoEwEkNJNOLeis:bmhxy9r) ([raw file](https://github.com/sillsdev/web-languageforge/blob/71fbe5dab97744574ff7829c2881037e50c43a5a/src/config.php.live#L)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

I'm assuming this is not needed anymore (same as the "fortest" version.  All of these environmentally changing values will be fed in via ENV vars.
</blockquote></details>

yes, you are correct - great to get rid of this debt.
___
*[src/Api/Library/Shared/CLI/RunClass.php, line 35 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsaE30yLLE3lUaN5y:-MY3pfnK5l7CQWHkV4F4:b-h2s0b8) ([raw file](https://github.com/sillsdev/web-languageforge/blob/6b5a12e0cd36148076784927d5aaf2a1eeffd820/src/Api/Library/Shared/CLI/RunClass.php#L35)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

not needed anymore.
</blockquote></details>

Actually, this is a special case for the LFMerge test suite in a separate repo which, honestly, probably won't work in the new Docker environment.  There's no way that you would have known that. The `RunClass.php` is basically a PHP binding for the LFMerge C# code to call PHP methods.  The production code, as well as the LFMerge tests apparently, make use of this PHP file.

So in this case SF_DATABASE still must be defined since it's not being run in the Docker environment.
___
*[test/app/setupTestEnvironment.php, line 36 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsrgyCpzzGTYXL5OP:-MY3t5Ps5_ZyrWgtenqJ:b-5mpkrk) ([raw file](https://github.com/sillsdev/web-languageforge/blob/6b5a12e0cd36148076784927d5aaf2a1eeffd820/test/app/setupTestEnvironment.php#L36)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

not sure this is even relevant...
</blockquote></details>

this is a last ditch failsafe condition before doing a "dropAllCollections()".  It's relevant, but I don't know that the condition has been exercised in years since it was likely put in place while we were early on in development and wanted to protect against a developer mistake.  I think it's fine to remove.  I approve.   
___
*[test/common/MongoTestEnvironment.php, line 40 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsx3H53L2v1HU0Xo8:-MY3tcXf2Ct5_xZKTgOL:bmkcjg7) ([raw file](https://github.com/sillsdev/web-languageforge/blob/6b5a12e0cd36148076784927d5aaf2a1eeffd820/test/common/MongoTestEnvironment.php#L40)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

I don't think this is possible.
</blockquote></details>

This probably isn't best practice now, but our system was designed to be able to run tests locally using a "test database" without interfering with other local projects/data, if that makes sense.

So this comment acknowledges the danger of unintentionally deleting non-test data.

Probably from the other comments I've made above you can see how scriptureforge_test is a valid database

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -157,6 +161,8 @@ services:
       - WAIT_HOSTS=db:27017, mail:25
       - "LDAPI_BASE_URL=http://ld-api:3000/api/v2/"
       - ENVIRONMENT=development
+      - DATABASE=scriptureforge
```
Comment (megahirt) [inline docker/docker-compose.yml:164]:
```suggestion
      - DATABASE=scriptureforge_test
```

**Review by megahirt — CHANGES_REQUESTED** [review]
These two databases should be set to `scriptureforge_test`
Thanks!

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -172,6 +178,8 @@ services:
     environment:
       - WAIT_HOSTS=db:27017, mail:25
       - ENVIRONMENT=development
+      - DATABASE=scriptureforge
```
Comment (megahirt) [inline docker/docker-compose.yml:181]:
```suggestion
      - DATABASE=scriptureforge_test
```

**Review by longrunningprocess — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/914)* status: 10 of 17 files reviewed, 6 unresolved discussions (waiting on @megahirt)

___
*[docker/docker-compose.yml, line 136 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MY3o0R-1EkLYDzXVtij:-MYBIMXz91OIQuRlQQaO:b2vdc6b) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L136)):*
<details><summary><i>Previously, megahirt (Christopher Hirt) wrote…</i></summary><blockquote>

I'm actually not certain that the DATABASE line is necessary.  Is it used anywhere in this container?  Or did you just add it for completeness?
</blockquote></details>

I just assumed it was needed but you're right, I took it out and it didn't seem to make a difference.  Obviously there's something here I'm missing.  Thanks for pointing it out.
___
*[docker/docker-compose.yml, line 164 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmrREk28bL46lCMZaU-r2-164:-MYBKu778_iv4BENSpkV:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L164)):*
<details><summary><i>Previously, megahirt (Christopher Hirt) wrote…</i></summary><blockquote>

```suggestion
      - DATABASE=scriptureforge_test
```
</blockquote></details>

Done.
___
*[docker/docker-compose.yml, line 164 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MY3nws6EAM2-30nQCC6:-MYBIgyr9MTIYRCq3LPO:b-12s092) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L164)):*
<details><summary><i>Previously, megahirt (Christopher Hirt) wrote…</i></summary><blockquote>

There is a separate database for running tests, `scriptureforge_test`.  I suggest an edit in github review.
</blockquote></details>

yep, no change in local e2e tests doing this
___
*[docker/docker-compose.yml, line 181 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmrREk28bL46lCMZaU-r2-181:-MYBKlSQ4L9O94mIESn9:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L181)):*
<details><summary><i>Previously, megahirt (Christopher Hirt) wrote…</i></summary><blockquote>

```suggestion
      - DATABASE=scriptureforge_test
```
</blockquote></details>

Done.
___
*[docker/docker-compose.yml, line 181 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MY3njSs2qNKzP0SbRU7:-MYBKJbeD6BSAEP3TGH5:bp29e36) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L181)):*
<details><summary><i>Previously, megahirt (Christopher Hirt) wrote…</i></summary><blockquote>

There is a separate database for running tests, `scriptureforge_test`.  I suggest an edit in github review.
</blockquote></details>

yep, this works fine for the unit tests locally.
___
*[docker/app-for-e2e/Dockerfile, line 4 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmr_deBqAiuyFggNAa:-MYBPDllBaP3UzI3tlHV:b-k1xhga) ([raw file](https://github.com/sillsdev/web-languageforge/blob/71fbe5dab97744574ff7829c2881037e50c43a5a/docker/app-for-e2e/Dockerfile#L4)):*
<details><summary><i>Previously, megahirt (Christopher Hirt) wrote…</i></summary><blockquote>

Since you've brought it up, I looked and only found 3 references to `TestMode`, all in `CommunicateHelper.php`.  The three instances don't seem to add value to me, and I wonder if we can simply remove the variable from the code and keep just "not test mode" case.  I suspect that we will be able to run the unit and e2e tests just fine without it.

So in conclusion, just delete references to TestMode all together.  It looks like the test config also defined the database name, which I've addressed above.  Since the database name is coming in via environment variable, then maybe you can really get rid of the config.fortest file all together - that would be sweet!
</blockquote></details>

removed it and unit tests are still good and the e2e tests seem unaffected locally. We'll see how TC feels about all this ;-)
___
*[src/Api/Library/Shared/CLI/RunClass.php, line 35 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsaE30yLLE3lUaN5y:-MYBQt3RABiK09lqa5xE:b-5ek78m) ([raw file](https://github.com/sillsdev/web-languageforge/blob/6b5a12e0cd36148076784927d5aaf2a1eeffd820/src/Api/Library/Shared/CLI/RunClass.php#L35)):*
<details><summary><i>Previously, megahirt (Christopher Hirt) wrote…</i></summary><blockquote>

Actually, this is a special case for the LFMerge test suite in a separate repo which, honestly, probably won't work in the new Docker environment.  There's no way that you would have known that. The `RunClass.php` is basically a PHP binding for the LFMerge C# code to call PHP methods.  The production code, as well as the LFMerge tests apparently, make use of this PHP file.

So in this case SF_DATABASE still must be defined since it's not being run in the Docker environment.
</blockquote></details>

ok, i"ll put this one back with a hardcoded value, please check to see if it's what you expect.
___
*[test/common/MongoTestEnvironment.php, line 40 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsx3H53L2v1HU0Xo8:-MYBSBNy4U3RBg8fvWM_:b-qzwxrb) ([raw file](https://github.com/sillsdev/web-languageforge/blob/6b5a12e0cd36148076784927d5aaf2a1eeffd820/test/common/MongoTestEnvironment.php#L40)):*
<details><summary><i>Previously, megahirt (Christopher Hirt) wrote…</i></summary><blockquote>

This probably isn't best practice now, but our system was designed to be able to run tests locally using a "test database" without interfering with other local projects/data, if that makes sense.

So this comment acknowledges the danger of unintentionally deleting non-test data.

Probably from the other comments I've made above you can see how scriptureforge_test is a valid database
</blockquote></details>

ok, I"ll put it back

**Review by longrunningprocess — COMMENTED** [review]
> Reviewed 9 of 16 files at r1, 1 of 4 files at r2, 7 of 7 files at r3.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/914)* status: all files reviewed, 6 unresolved discussions (waiting on @megahirt)

___
*[src/Api/Library/Shared/CLI/RunClass.php, line 35 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsaE30yLLE3lUaN5y:-MYFi-Hu4ar1rPzSILHG:b-f9tsbz) ([raw file](https://github.com/sillsdev/web-languageforge/blob/6b5a12e0cd36148076784927d5aaf2a1eeffd820/src/Api/Library/Shared/CLI/RunClass.php#L35)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

ok, i"ll put this one back with a hardcoded value, please check to see if it's what you expect.
</blockquote></details>

since putting it back broke a unit test, I'm going to go ahead and remove it for now so I can fold this thing into `master` and work on the k8s runtime.  As we're working through functionality in the QA env we can work through these types of special use cases and figure out what we need to do for each.  Thanks for pointing this out, let's plan to resolve this is a later PR when it's blocking something.

**Review by longrunningprocess — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-:-MYFimeI0hBpIu7VhdDR:bnfp4nl)* status: all files reviewed, 6 unresolved discussions (waiting on @megahirt)

**Review by longrunningprocess — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/914)* status: all files reviewed, 5 unresolved discussions (waiting on @megahirt)

___
*[docker/docker-compose.yml, line 136 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MY3o0R-1EkLYDzXVtij:-MYFj0a1-qvMMsQjfJ5s:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L136)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

I just assumed it was needed but you're right, I took it out and it didn't seem to make a difference.  Obviously there's something here I'm missing.  Thanks for pointing it out.
</blockquote></details>

Done.
___
*[docker/docker-compose.yml, line 164 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MY3nws6EAM2-30nQCC6:-MYFj1Fe11ThX7PfXatu:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L164)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

yep, no change in local e2e tests doing this
</blockquote></details>

Done.
___
*[docker/docker-compose.yml, line 164 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmrREk28bL46lCMZaU-r2-164:-MYFj1bQC8J_jEhzzw-r:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L164)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.
___
*[docker/docker-compose.yml, line 181 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MY3njSs2qNKzP0SbRU7:-MYFj2BY48fM3U_vIvTO:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L181)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

yep, this works fine for the unit tests locally.
</blockquote></details>

Done.
___
*[docker/docker-compose.yml, line 181 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmrREk28bL46lCMZaU-r2-181:-MYFj2wyCKMoJdfqyBW5:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L181)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

**Review by longrunningprocess — COMMENTED** [review]
> Dismissed @megahirt from 5 discussions.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/914)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @longrunningprocess)

___
*[docker/docker-compose.yml, line 136 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MY3o0R-1EkLYDzXVtij:-MYFjFY_9JzRcVID4lPT:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L136)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.
___
*[docker/docker-compose.yml, line 164 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MY3nws6EAM2-30nQCC6:-MYFjHL53ZB88hHbFP-r:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L164)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.
___
*[docker/docker-compose.yml, line 164 at r2](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmrREk28bL46lCMZaU-r2-164:-MYFjIEN1im9dHtSlKWj:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/4a7bde8a20d5361ccb73cb6c6ae7b775fb12123d/docker/docker-compose.yml#L164)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

**Review by megahirt — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/914)* status: :shipit: complete! all files reviewed, all discussions resolved

___
*[src/Api/Library/Shared/CLI/RunClass.php, line 35 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsaE30yLLE3lUaN5y:-MYKdGhUAoNgcMiguYmG:bnmzpe7) ([raw file](https://github.com/sillsdev/web-languageforge/blob/6b5a12e0cd36148076784927d5aaf2a1eeffd820/src/Api/Library/Shared/CLI/RunClass.php#L35)):*
<details><summary><i>Previously, longrunningprocess (billy clark) wrote…</i></summary><blockquote>

since putting it back broke a unit test, I'm going to go ahead and remove it for now so I can fold this thing into `master` and work on the k8s runtime.  As we're working through functionality in the QA env we can work through these types of special use cases and figure out what we need to do for each.  Thanks for pointing this out, let's plan to resolve this is a later PR when it's blocking something.
</blockquote></details>

you should still have the line
`define('SF_DATABASE', 'scriptureforge_test');`
Yes, the unit tests will pass with out it but the LFMerge unit tests, if they ever get run again, will fail.  That's in a separate repo but depends upon this file.

**Review by longrunningprocess — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/914)* status: :shipit: complete! all files reviewed, all discussions resolved

___
*[src/Api/Library/Shared/CLI/RunClass.php, line 35 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/914#-MXmsaE30yLLE3lUaN5y:-MYM7q5wAo_aPXWFDVmX:bj2kt59) ([raw file](https://github.com/sillsdev/web-languageforge/blob/6b5a12e0cd36148076784927d5aaf2a1eeffd820/src/Api/Library/Shared/CLI/RunClass.php#L35)):*
<details><summary><i>Previously, megahirt (Christopher Hirt) wrote…</i></summary><blockquote>

you should still have the line
`define('SF_DATABASE', 'scriptureforge_test');`
Yes, the unit tests will pass with out it but the LFMerge unit tests, if they ever get run again, will fail.  That's in a separate repo but depends upon this file.
</blockquote></details>

ok, how would you feel about adding this back in when it comes time to get the LFMerge unit tests running?

**Pull request merged**